photon-server/src/main/java/org/photonvision/vision/camera/USBCameraSource.java

```diff
@@ -80,17 +91,40 @@ protected USBCameraSettables(CameraConfiguration configuration) {
             calculateFrameStaticProps();
         }
 
+        private int timeToPiCamV2RawExposure(double time_us) {
+            int retVal =
+                    (int) Math.round(time_us / 100.0); // PiCamV2 needs exposure time in units of 100us/bit
+            return Math.min(Math.max(retVal, 1), 10000); // Cap to allowable range for parameter
+        }
+
+        private double pctToExposureTimeUs(double pct_in) {
+            // Mirror the photonvision raspicam driver's algorithm for picking an exposure time
+            // from a 0-100% input
+            final double PADDING_LOW_US = 100;
+            final double PADDING_HIGH_US = 200;
+            return PADDING_LOW_US
+                    + (pct_in / 100.0) * ((1e6 / (double) camera.getVideoMode().fps) - PADDING_HIGH_US);
+        }
+
         @Override
-        public void setExposure(int exposure) {
+        public void setExposure(double exposure) {
             try {
+                int scaledExposure = 1;
                 if (cameraQuirks.hasQuirk(CameraQuirk.PiCam)) {
+                    camera.getProperty("white_balance_auto_preset").set(2); // Auto white-balance off
                     camera.getProperty("auto_exposure").set(1); // auto exposure off
-                    camera
-                            .getProperty("exposure_time_absolute")
-                            .set(MathUtils.safeDivide(10000, exposure)); // exposure time is a range, 0-10000
+
+                    scaledExposure =
+                            (int) Math.round(timeToPiCamV2RawExposure(pctToExposureTimeUs(exposure)));
+                    logger.info("Setting camera raw exposure to " + Integer.toString(scaledExposure));
```

Comment:
@Bankst should this be a debug-level log?

Reply:
Yes